python/activator/middleware_interface.py

```diff
@@ -838,39 +822,45 @@ def query_calibs_by_date(butler, label):
             all_callback=self._mark_dataset_usage,
         ))
         if calibs:
-            for dataset_type, n_datasets in self._count_by_key(calibs, lambda ref: ref.datasetType.name):
-                _log.debug("Found %d new calib datasets of type '%s'.", n_datasets, dataset_type)
-        else:
-            _log.debug("Found 0 new calib datasets.")
+            _log.debug("Found %d new calib datasets of type '%s'.", len(calibs), dataset_type.name)
         return calibs
 
-    def _export_ml_models(self):
-        """Identify the pretrained machine learning models to export from the
-        central butler.
+    def _export_generic(self, dataset_type, detector_id, physical_filter):
+        """Identify datasets to export from the central butler.
+
+        Parameters
+        ----------
+        dataset_type : `lsst.daf.butler.DatasetType`
+            The type of dataset to search for.
+        detector_id : `int`
+            Identifier of the detector to load calibs for.
+        physical_filter : `str`
+            Physical filter name of the upcoming visit. May be empty to indicate
+            no specific filter.
 
         Returns
         -------
-        models : iterable [`lsst.daf.butler.DatasetRef`]
+        datasets : iterable [`lsst.daf.butler.DatasetRef`]
             The datasets to be exported, after any filtering.
         """
-        # TODO: the dataset type name is subject to change (especially if more
-        # kinds of models are added in the future). Hardcoded names should
-        # become unnecessary with DM-40245.
-        try:
-            models = set(_filter_datasets(
-                self.central_butler, self.butler,
-                _generic_query(["pretrainedModelPackage"],
-                               collections=self._collection_ml_model,
-                               find_first=True,
-                               ),
-                all_callback=self._mark_dataset_usage,
-            ))
-        except _MissingDatasetError as err:
-            _log.error("Could not load ML models: %s", err)
-            models = set()
-        else:
-            _log.debug("Found %d new ML model datasets.", len(models))
-        return models
+        data_id = {"instrument": self.instrument.getName(),
+                   "skymap": self.skymap_name,
```

Comment:
I didn't realize that irrelevant dataIds are just ignored by Butler, and just learnt it. Do we worry that it may change in the future? A quick search in doc didn't seem to tell me if it is meant to stay that way.  Fine to me to just leave it. It does seem to be a convenient "feature" of Butler.